Disclaimer pop up

leenlab2 opened the pull request:
<!--- Provide a summary of your changes in the Pull Request Title above. -->
<!--- If this is a work in progress (not yet ready to be merged), make this a draft pull request. -->
Changed the disclaimer into a dismissable popup box right below the menu bar on all pages.

## Motivation and Context

<!--- Why is this pull request required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here: -->
<!--- https://docs.github.com/en/github/managing-your-work-on-github/managing-your-work-with-issues-and-pull-requests/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword -->
The disclaimer used to take up too much unnecessary space on the page. This PR is part of a set of layout changes detailed in Issue #1218.

## Your Changes

<!--- Describe your changes here. -->
**Description**:

- Removed the old disclaimer
- Resized container to account for removed disclaimer
- Created a new Disclaimer react component
- Inserted that component into `<Container />`, `<Grid />`, and `Generate`.
- Increased the `z-index` of `.overlay` to hide the disclaimer
- Changed link colors on the disclaimer to account for the different background color
- Added a 'Do Not Show Again' checkbox to stop the disclaimer from showing up every time

**Type of change** (select all that apply):
<!--- Put an `x` in all the boxes that apply. -->
<!--- Remove any lines that do not apply. -->

- [x] Other: User interface changes

## Testing

<!--- Please describe in detail how you tested this pull request. -->
<!--- This can include tests you added and manual testing in the web browser. -->
Manual testing in browser.

## Questions and Comments (if applicable)

<!-- Ask any questions you have for the maintainers of this project regarding this PR. -->
<!-- Please describe the steps you have already taken to find the answer to your question. -->
<!-- This will ensure that we can give you clear and relevant advice. -->
<!-- If you have additional comments add them here as well. -->
None!

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the CircleCI tests have
  passed. <!-- (check after opening pull request) -->

leenlab2 commented:
Comparison screenshots to showcase changes:

Before
![image](https://user-images.githubusercontent.com/74102544/123870809-f0262500-d943-11eb-9be1-dc91fa7be9a2.png)

After
![image](https://user-images.githubusercontent.com/74102544/123871158-675bb900-d944-11eb-8c28-d3d4091fac10.png)

leenlab2 commented:
Alright, I'll get working on it! I used ids instead of classes for some since they were already present in the css file (eg #disclaimerDiv), but I'll work on improving that.

david-yz-liu commented on js/components/common/Disclaimer.js:
```diff
@@ -0,0 +1,44 @@
+    constructor(props) {
+        super(props);
+        this.state = {
+            dontShowDisclaimer: localStorage.getItem('DontShowDisclaimer')
```
The names here could be better. The state name should be something standard like `hidden`, and the local storage name a corresponding `hide-disclaimer`.

david-yz-liu replied:
Oh also, do the conversion to a boolean here rather than in the `render`. Logically, this state should be a bool, not a string.

david-yz-liu commented on js/components/common/Disclaimer.js:
```diff
@@ -0,0 +1,44 @@
+        }
+    }
+
+    handleClose = () => {
```
Avoid modifying the DOM directly, since we want to use React to do this for us. This method should set the "hide disclaimer" state to `true`. You don't need the `dismiss` class or styling, I don't think the transition is worth it.

david-yz-liu commented on js/components/common/Disclaimer.js:
```diff
@@ -0,0 +1,44 @@
+        disclaimer.classList.toggle("dismiss");
+    }
+
+    handleCheck = () => {
```
The checkbox can be both selected and deselected. So you need to access the event object to see whether the box is checked or unchecked, as this determines what to set the local storage value to.

david-yz-liu commented on js/components/common/Disclaimer.js:
```diff
@@ -0,0 +1,44 @@
+
+    render() {
+        const timetable = <a href="https://timetable.iit.artsci.utoronto.ca/">Official Timetable</a>;
+        const calendar = <a href="http://calendar.artsci.utoronto.ca/">Calendar</a>;
```
It looks like the new permanent URL is `https://artsci.calendar.utoronto.ca/`, so please update to use this. Also, please change "Calendar" to "Academic Calendar".

david-yz-liu commented on js/components/common/Disclaimer.js:
```diff
@@ -0,0 +1,44 @@
+            )
+        }
+    }
+}
```
Make sure to end this file with a blank line.

david-yz-liu commented on js/components/common/Disclaimer.js:
```diff
@@ -0,0 +1,44 @@
+                    </label>
+                    <button id="closeDisclaimer" class="close-banner" onClick={this.handleClose}>&times;</button>
+                </div>
+            )
```
Add a semicolon here.

david-yz-liu commented on js/components/common/Disclaimer.js:
```diff
@@ -0,0 +1,44 @@
+            return (null);
+        } else {
+            return (
+                <div id="disclaimerDiv" className="popup-banner">
```
Since you've added the classes, I suspect that the ids here and for the inner input/button elements are no longer necessary, and you can remove them.

david-yz-liu commented on js/components/grid/grid.js.jsx:
```diff
@@ -164,8 +165,10 @@ class Grid extends React.Component {
 
   render() {
     const updatedList = this.state.hoveredLecture ? this.state.selectedLectures.concat(this.state.hoveredLecture) : this.state.selectedLectures;
+
```
Revert this change

david-yz-liu commented on style/app.scss:
```diff
@@ -233,10 +233,51 @@ a:active
-  font-size : 11pt;
+  font-size        : 11pt;
+  position         : absolute;
+  background-color : #ffdd61;
```
I'd prefer a yellow that's slightly less saturated (i.e., closer to white).

david-yz-liu commented on style/app.scss:
```diff
@@ -233,10 +233,51 @@ a:active
+
+.popup-banner a
+{
+  color : #af16af;
```
Don't set a different colour here. Instead, use `text-decoration: underline` (in fact, we should have `text-decoration: underline` active by default for all `a` tags, but that's a different story).

david-yz-liu commented on style/app.scss:
```diff
@@ -233,10 +233,51 @@ a:active
+{
+  font-weight : lighter;
+  color       : #696969;
+  position    : absolute;
```
Don't overuse absolute positioning. In this particular case, you shouldn't assume that there will be empty space in the bottom-right corner of the banner, as this depends on screen size. Use a different layout here. For example, put the "X" in the top-right corner, and the checkbox at the bottom.

david-yz-liu commented on style/app.scss:
```diff
@@ -233,10 +233,51 @@ a:active
+
+.close-banner
+{
+  cursor      : pointer;
```
This rule is redundant

david-yz-liu commented on style/app.scss:
```diff
@@ -233,10 +233,51 @@ a:active
+
+.popup-banner label
+{
+  font-weight : lighter;
```
In general, do not introduce new styles for text or other elements unless there's a good reason we need something new. In this case, you should remove this rule and the new color.

david-yz-liu reviewed (commented):
In addition to the comments I left inline, I'd like you to experiment with different placements of this notification. I'm not sure a full-width banner is the best approach. You can try making the banner less wide and either centered at the top, centered at the bottom, or in a corner (I've seen bottom-left or bottom-right on a few websites). Possibly with a stronger border and/or rounded edges to make it more obvious that it's "on top of" the rest of the page.

leenlab2 commented:
I fixed the issues you asked for, and changed the disclaimer design to look like this:
![image](https://user-images.githubusercontent.com/74102544/124363448-5664bd80-dc4c-11eb-93e1-f91c6fac3528.png)

david-yz-liu reviewed (commented):
@leenlab2 very nice work! I just left a few more comments. Almost there!

leenlab2 commented:
Alright, should be done now! Let me know if there is anything else I need to fix.
